### PR TITLE
story/ITE-4/FE_sign_up Created signup form

### PR DESCRIPTION
**Pull request by akhilarjun3-sureify**
output-screenshot
output-screenshot2
output-screenshot3

### CODE REVIEW

File: src/Components/SignupPage.jsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        } else if (value != signUpForm.password) {
+          message = "password not match";
+        }
+        break;
+      case "mobile":
+        if (!value) {
+          message = " mobile no. is required";
+        } else if (!/^\d{10}$/.test(value)) {
+          message = "mobile no. should have 10 digits";
+        } else if (!/^[9,8,7,6]\d{9}$/.test(value)) {
```
Comment (Madhukar-Sureify) [inline src/Components/SignupPage.jsx:133]: what is user mobile number starts with 22  or 11 few states in us their mob num starts with 1, 2  etc

File: src/Components/SignupPage.jsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+          message = "Email is required";
+        } else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value)) {
+          message = "Email is not valid";
+        }
+        break;
+      case "password":
+        if (!value) {
+          message = "Password is required";
+        } else if (value.length < 8) {
+          message = "password length should be atleast 8";
```
Comment (Madhukar-Sureify) [inline src/Components/SignupPage.jsx:113]: All the error messages should be in sentence case

File: src/App.js [hunk 3]
```diff
@@ -1,25 +1,12 @@
...
-        </a>
-      </header>
-    </div>
-  );
-}
-
-export default App;
+ import { Box, useTheme } from '@mui/material'  
+import React from 'react';
+import SignupPage from './Components/SignupPage';
```
Comment (abhishekdundigalla-sureify) [inline src/App.js:3]: Folder name must start with lower case (change Components to components)

File: src/Components/SignupPage.jsx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+  IconButton,
+  InputAdornment,
+  Paper,
+  TextField,
+  Toolbar,
+  Typography,
+} from "@mui/material";
+
+import React, { useEffect, useState } from "react";
+var message = "";
```
Comment (abhishekdundigalla-sureify) [inline src/Components/SignupPage.jsx:18]: use let instead of var

File: src/Components/SignupPage.jsx [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+const paperStyle = {
+  padding: "15px",
+  margin: "20px 0px",
+};
+const textField = {
+  margin: "8px 0px",
+};
```
Comment (abhishekdundigalla-sureify) [inline src/Components/SignupPage.jsx:26]: use styled components for styles
folder structure : 
- Signup
   - SignupPage.jsx
   - SignupStyles.js 
write all the styled components in the SignupStyles.js, export from there and use it in SignupPage component

File: src/Components/SignupPage.jsx [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+const SignupPage = () => {
+  const theme = useTheme();
+  const [signUpForm, setSignUpForm] = useState(initialize());
+  const [error, setErrors] = useState({});
+  const [submitStatus, setSubmitStatus] = useState("");
+  const [seePassword, setSeePassword] = useState(false);
+
+  const sendSignUpDataToServer = async (data) => {
+    console.log(data,"what a drag")
```
Comment (abhishekdundigalla-sureify) [inline src/Components/SignupPage.jsx:44]: remove logs

File: src/Components/SignupPage.jsx [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+    try {
+      delete data.cpassword
+      const res = await fetch(signUrl, {
+        method: "POST",
+        headers: { "Content-Type": 'application/json' },
+        body:JSON.stringify(data)
+      })
+      if (res.ok) {
+        alert("submitted")
```
Comment (abhishekdundigalla-sureify) [inline src/Components/SignupPage.jsx:53]: remove alerts

File: src/components/SignupPage.jsx [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+  Paper,
+  TextField,
+  Toolbar,
+  Typography,
+} from "@mui/material";
+
+import React, { useEffect, useState } from "react";
+import { SignUpTextField } from "./styles";
+import image from "../images/image.png";
+import image2 from "../images/image2.png";
```
Comment (harshavardhan-sureify) [inline src/components/SignupPage.jsx:21]: name the image according to its usage

File: src/components/SignupPage.jsx [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+              sx={{
+                backgroundColor: theme.palette.success.main,
+                marginTop: "8px",
+              }}
+            ></Avatar>
+            <Typography variant="h5">Sign up</Typography>
+            <Typography variant="caption">
+              Create your CodeByte Account
+            </Typography>
+            <form method="post" onSubmit={handleSubmit}>
```
Comment (harshavardhan-sureify) [inline src/components/SignupPage.jsx:172]: can remove method post alrready defined in fetch call

**Review by maniknr-sureify — COMMENTED** [review]
From Next time,Use comment lines for better understanding

**Review by maniknr-sureify — COMMENTED** [review]
From Next time,Use comment lines for better understanding

**Review by harshavardhan-sureify — APPROVED** [review]

File: src/Components/SignupPage.jsx [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+          message = "Name should have atleast length 3";
+        } else {
+          message = "";
+        }
+        break;
+
+      case "email":
+        if (!value) {
+          message = "Email is required";
+        } else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value)) {
```
[the remark this answers is not in the document]
Comment (akhilarjun3-sureify) [reply src/Components/SignupPage.jsx]: It is also correct .For better understanding modified it to /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/

File: src/Components/SignupPage.jsx (hunk 1 above)
Comment (akhilarjun3-sureify) [reply src/Components/SignupPage.jsx]: Removed mobile number as it is not necessary.

File: src/Components/SignupPage.jsx (hunk 2 above)
Comment (akhilarjun3-sureify) [reply src/Components/SignupPage.jsx]: Yes changed .

File: src/Components/SignupPage.jsx (hunk 4 above)
Comment (akhilarjun3-sureify) [reply src/Components/SignupPage.jsx]: used let instead of var.

File: src/Components/SignupPage.jsx (hunk 6 above)
Comment (akhilarjun3-sureify) [reply src/Components/SignupPage.jsx]: Removed all logs

File: src/constants.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
+const goBaseUrl = "http://localhost:8080/"
+const phpBaseUrl = "http://localhost/";
+const prefix = ""
+
+const signUrl = goBaseUrl +prefix+"signup";
```
[the remark this answers is not in the document]
Comment (akhilarjun3-sureify) [reply src/constants.js]: Yes.Modified  constants.js

File: src/components/SignupPage.jsx (hunk 8 above)
Comment (akhilarjun3-sureify) [reply src/components/SignupPage.jsx]: used logo,ProfileImage for respective images

File: src/Components/SignupPage.jsx (hunk 7 above)
Comment (akhilarjun3-sureify) [reply src/Components/SignupPage.jsx]: Removed all alerts

File: src/components/SignupPage.jsx (hunk 9 above)
Comment (akhilarjun3-sureify) [reply src/components/SignupPage.jsx]: Yes removed.

File: src/App.js (hunk 3 above)
Comment (akhilarjun3-sureify) [reply src/App.js]: changed the folder names

**Review by harshavardhan-sureify — APPROVED** [review]

File: src/components/SignupPage.jsx [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+
+  const sendSignUpDataToServer = async (data) => {
+    try {
+      delete data.cpassword;
+      const res = await fetch(GO_BASE_URL, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(data),
+      });
+      if (res.ok) {
```
Comment (abhishekdundigalla-sureify) [inline src/components/SignupPage.jsx:49]: if condition can be removed if you are not using it for any further actions

**Review by abhishekdundigalla-sureify — APPROVED** [review]

**Review by Madhukar-Sureify — APPROVED** [review]

**Pull request merged**